docs/code-organization.md

```diff
@@ -0,0 +1,45 @@
+# Code organization proposal
+
+Thoughts on reorganizing existing codebase:
+- Move types and functions shared between SMP Server and SMP Agent to `Simplex/Messaging/Common`. Shared types are mostly types from `Server/Transmission` used in Agent and `Agent/Store`. Shared functions are SMP parsers. Maybe something else?
+- Separate parsers and serializers from Transmission, moving them into separate modules, e.g. for agent related ones to `Simplex/Messaging/Agent/SerDe.hs`?
+- For bigger modules decrease amount of functions per module, moving some inner (`where`) and reused functions to submodules, for example for Store utility methods to `Store/Utils.hs`, probably for Agent - to `Agent/Commands.hs`.
+- Move modules under similarly named folders? E.g. `Store.hs` under `Simplex/Messaging/Store`, `SQLite.hs` under `Simplex/Messaging/Store/SQLite`, etc. Same for server modules.
```

Comment:
What we have is conventional for all Haskell packages, otherwise you'd have Simplex.Messaging.Store.Store - better keep as is.